### ISSUE TITLE
Asset checked out to user without a location retains default location

### ISSUE THREAD

**Issue by wurzelton** [grokability/snipe-it#15997]
**Description of issue:**
When an asset is checked out to a user who does not have an assigned location (i.e., the location is null), the asset’s location field still inherits the “default location” instead of being set to null (as derived from the user).
<img width="593" alt="wrongly derived from default location" src="https://github.com/user-attachments/assets/05a56a93-3bcf-4745-be59-0bda827df892" />

**Why this is a problem:**
As the IT department, we do not track the physical locations of our users. When an asset is assigned to a user, we only know that the user is responsible for it, but we have no knowledge of the device’s actual location. However, we know for certain that the device is not in the “default location” (which is usually our IT office). The current behavior creates confusion, as the location field incorrectly suggests that the asset remains in the default location.

**Workaround with default-user-location not working neither:**
As a workaround, we attempted to assign a default location called “user-specific” to all users. The idea was that this value would populate the asset’s location field when the asset is checked out to a user. To implement this, we used the ldap-sync command:
php artisan snipeit:ldap-sync --location=user-specific --summary

While this workaround correctly sets the location to “user-specific” for assets checked out to users, it also incorrectly assigns “user-specific” to assets checked out to a specific location. This behavior is problematic because assets explicitly checked out to a location should retain that location in the location field, not “user-specific.”

<img width="580" alt="ldap-sync sets location wrong" src="https://github.com/user-attachments/assets/a58f9646-f1d3-41b5-88dd-d3b55392d014" />

If this second issue (the incorrect overwrite of location-specific assets with “user-specific”) could be resolved, it might reduce the need to address the first issue for us. However, it would still be beneficial to understand the logic behind the current behavior, in case there is a deliberate reason for it.

**Comment by snipe** [comment]
I think the challenge here is that if we do anything weird here, we'll break functionality for folks with different workflows. There are countless issues and discussions here talking about individual use-cases and how folks use locations (both "real" and "default") and every time we try to fix it for one set of people, we end up breaking it for others. :(

**Comment by wurzelton** [comment]
Thank you very much for your response. Even though the current behavior doesn’t make sense to me from a logical perspective, I understand your point that changing it might inevitably disrupt how others are using it individually.

In that case, I’d like to simply report the second issue I described, which, to me, seems like a bug (please correct me if I’m wrong). Specifically, the fact that during the LDAP sync with the --location=user-specific flag, the location field of assets checked out to a location (and not a user) also gets overwritten with “user-specific.”

php artisan snipeit:ldap-sync --location=user-specific --summary
<img width="580" alt="398200971-a58f9646-f1d3-41b5-88dd-d3b55392d014" src="https://github.com/user-attachments/assets/d2df98a1-ce5d-4c33-95e5-f766a3a3e2d9" />


Should I create a new issue for this?

**Comment by snipe** [comment]
Yeah, if you could create a second issue for that, that would be helpful. (I'm not really sure why assets assigned to locations would be included in an ldap sync anyway, since LDAP doesn't have locations as a first class object.)

**Comment by snipe** [comment]
Like, this line here should only be affecting assets specifically that are checked out to users:

https://github.com/snipe/snipe-it/blob/c3d52af5461953b6bcbd49072ea22e32341405d8/app/Console/Commands/LdapSync.php#L429-L430

(The `assigned_type` part there is the key)

**Comment by uberbrady** [comment]
I think you were on the right path with the "user-specific" Location (or, you could say "remote" if you wanted).

But once you had all of those users assigned to the "user-specific" location, you would've wanted to run a _regular_ LDAP sync - no flags involved. That should move all of your assets over to the user's new Location ("user-specific", when appropriate) and it would've worked how you want. The `--location` flag doesn't work the way you're expecting; it's more often used when someone says "sync this BaseDN over to this Location" - specifically when dealing with many different BaseDN's all to the same Location. (There is also some stuff you could do, depending on how your directory looks, where you *could* just sync one BaseDN of all of your remote users to one specific location, and you can do that all within Snipe-IT with an "LDAP OU Override" in the location in question, no flags necessary).

But the question of moving assets from the "default location" to the `NULL` location is a good one, and I think I agree with @snipe that changing the behavior there is going to make 50% of users super-happy, and 50% super-miserable (which is always the challenge with projects like ours).

Hope that makes sense, and thank you for the report!

**Comment by wurzelton** [comment]
Dear both,

Thank you both for your responses. I have now created a new issue (#16032) for the bug where the LDAP sync also changes the location field of assets assigned to locations.
As you mentioned, @snipe, this behavior shouldn’t occur according to the code snippet you referenced, but I am able to reproduce it in my setup (details are in the ticket).

@uberbrady, even if I no longer include the --location flag in subsequent LDAP syncs, assets assigned to locations should not have been affected during the initial LDAP sync with the --location flag either.
And since we want to ensure that any new users are automatically assigned the location “user-specific” (as we never assign locations to users otherwise), we will likely run the LDAP sync regularly with the --location flag.

Now that there is a new ticket for the LDAP sync issue, and the issue regarding “moving assets from the ‘default location’ to the NULL location” will not be implemented, I believe this ticket can be closed.

**Comment by snipe** [comment]
Ping, @uberbrady
